### PR TITLE
xeval/evaluator: implement json_type

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,
    This PR implement function `json_type` in evaluator, the related PR in `tidb` is https://github.com/pingcap/tidb/pull/3492,
and the related source files in `tidb` are

- https://github.com/pingcap/tidb/blob/master/expression/builtin_json.go
- https://github.com/pingcap/tidb/blob/master/distsql/xeval/eval_json.go
- https://github.com/pingcap/tidb/blob/master/distsql/xeval/eval.go

And the file datum.rs is the same with https://github.com/pingcap/tikv/pull/1996. we should do review after it has been merged.

- [x] https://github.com/pingcap/tikv/pull/1996

@hhkbp2 @hicqu @lishihai9017 @andelf PTAL

### CODE REVIEW

**Comment by hhkbp2** [comment]
LGTM

**Comment by lishihai9017** [comment]
LGTM

File: src/util/codec/datum.rs [hunk 1]
```diff
@@ -1341,4 +1395,55 @@ mod test {
...
+
+    #[test]
+    fn test_cast_as_json() {
+        let tests = vec![
+            (Datum::I64(1), "1.0"),
+            (Datum::F64(3.3),"3.3"),
+            (Datum::Bytes(br#""Hello,world""#.to_vec()), r#""Hello,world""#),
+            (Datum::Bytes(b"[1, 2, 3]".to_vec()), "[1, 2, 3]"),
+            (Datum::Bytes(b"{}".to_vec()), "{}"),
+            (Datum::I64(1), "true"),
```
Comment (hicqu) [inline src/util/codec/datum.rs]:
```sql
SELECT CAST(NULL as JSON);
```
should returns `NULL`.

**Comment by hicqu** [comment]
LGTM.

File: src/util/codec/datum.rs (hunk 1 above)
Comment (AndreMouche) [inline src/util/codec/datum.rs]: It's the same with `tidb` https://github.com/pingcap/tidb/blob/master/util/types/datum.go#L1227 @hicqu

File: src/util/codec/datum.rs [hunk 2]
```diff
@@ -460,7 +460,7 @@ impl Datum {
         };
         parse_json_path_expr(v)
     }
-
+
```
Comment (hicqu) [inline src/util/codec/datum.rs]: remove this line.

**Pull request merged**